services/doctors/routers/agenda.py

```diff
@@ -45,3 +45,25 @@ async def post_agenda(
             'User not authorized to access this resource'
         )
 
+
+# Delete doctor agenda
+@router.delete("/doctors/agenda/{agenda_id}")
+async def delete_agenda(
+    agenda_id: int,
+    current_user: User = Depends(Authentication().get_doctor) 
+):
+
+    if current_user:
+
+        
+        try:
+            AgendaManager().delete(agenda_id,current_user.id)
+        except Exception as e:
+            print(e)
+            return CustomError().error_500(
+                "Error when deleting the agenda")
```

Comment:
**Suggestion:** The exception handling in the `delete_agenda` function is too broad. It's a good practice to catch specific exceptions that you expect might be raised in the try block. This way, you can provide more informative error messages and handle different types of exceptions differently if needed.
```suggestion
        try:
            AgendaManager().delete(agenda_id,current_user.id)
        except ValueError as e:
            print(e)
            return CustomError().error_400(str(e))
        except Exception as e:
            print(e)
            return CustomError().error_500(
                "Unexpected error when deleting the agenda")
```